main.cpp

```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
+}
+
+// ================ Elevator Functions ===================
+
+Elevator::Elevator(std::string elevatorName) : maxWeight(800.0), currentFloor(0), name(elevatorName)
+{
+    std::cout << "****** Elevator Created & Ready for Service ******" << std::endl;
+}
+
+Elevator::~Elevator()
+{
+    std::cout << "****** Elevator " << name << " Destroyed & Out of Service ******" << std::endl;
+}
+
+void Elevator::goToFloor(int designatedFloor)
+{   
+    if(designatedFloor <= highestFloor && designatedFloor >= lowestFloor)
+    {
+        std::cout << "On the way to floor " << designatedFloor << std::endl;	
+
+        while(designatedFloor != currentFloor)
+            (designatedFloor > currentFloor) ? ++currentFloor : --currentFloor; 	
+
+        elevatorArrived = true;
+
+        std::cout << "Elevator " << name << " arrived to floor " << currentFloor << std::endl;
+    }
+    else
+    {
+        elevatorArrived = false;
+        std::cout << "Floor " << designatedFloor << " does not exist!" << std::endl;
+    }
+}
+
+void Elevator::onboardPassenger(double userWeight)
+{
+    if(arrived() && canTakePassengerWeight(userWeight))
+        updateWeightCarried(userWeight, true);
+    else
+        std::cout << "This passenger cannot board the elevator" << std::endl;
+}
+
+void Elevator::offboardPassenger()
+{
+    if(arrived() && lastUserWeight > 0)
+        updateWeightCarried(lastUserWeight, false);
+    else
+        std::cout << "The elevator has not arrived to a floor yet or no passengers inside" << std::endl;
+}
+
+void Elevator::updateWeightCarried(double userWeight, bool goingInside)
+{
+    if(goingInside)
+    {
+        lastUserWeight = userWeight;
+        totalWeightCarried += lastUserWeight;
+        std::cout << "Passenger onboarded" << std::endl;
+    }
+    else
+    {
+        totalWeightCarried -= lastUserWeight;
+        std::cout << "Passenger offboarded" << std::endl;
+    }
+}
+
+bool Elevator::canTakePassengerWeight(double userWeight)
+{
+    if((userWeight + totalWeightCarried) > maxWeight)
+        return false;
+        
+    return true;
+}
+
+bool Elevator::arrived()
+{
+    return elevatorArrived;
+}
+
+// ============= SchoolBag Functions ===============
+
+SchoolBag::SchoolBag()
+{
+    std::cout << "****** SchoolBag Created ******" << std::endl;
+}
+
+SchoolBag::~SchoolBag()
+{
+    unpackPencilCase();
+    std::cout << "****** SchoolBag Destroyed ******" << std::endl;
+}
+
+void SchoolBag::packPencilCase()
+{
+    if(!schoolPencilCase.inSchoolBag && !schoolPencilCase.isOpen)
+    {
+        schoolPencilCase.inSchoolBag = true;
+        std::cout << "Packing the pencilcase... Good to go!" << std::endl;
+    }
+    else if(schoolPencilCase.inSchoolBag)
+        std::cout << "Pencilcase already packed" << std::endl;
+    else
+    {
+        schoolPencilCase.removeItems(schoolPencilCase.maxItemsToHold);
+        std::cout << "** You forgot to close your pencilcase!! **" << std::endl;
+    }
+}
```

Comment:
If you use curly braces for the `if()` block, use them for the `else if` and `else` blocks too.
- [ ] revise